packages/react-components/src/components/trace-context-component.tsx

```diff
@@ -193,13 +198,61 @@ export class TraceContextComponent extends React.Component<TraceContextProps, Tr
         this.props.messageManager.removeStatusMessage(this.INDEXING_STATUS_BAR_KEY);
         this.props.messageManager.removeStatusMessage(this.TIME_SELECTION_STATUS_BAR_KEY);
         this.props.removeResizeHandler(this.onResize);
+        signalManager().off(Signals.TIMEGRAPH_ZOOMED, this.onTimeGraphZoomed);
+        signalManager().off(Signals.TIMEGRAPH_RESET, this.onTimeGraphReset);
     }
 
     async componentDidUpdate(): Promise<void> {
         // Rebuild enables tooltip on newly added output component
         ReactTooltip.rebuild();
     }
 
+    private doHandleTimeGraphZoomedSignal(hasZoomedIn: boolean) {
+        if (hasZoomedIn) {
+            this.zoomButton(hasZoomedIn);
+        } else {
+            this.zoomButton(hasZoomedIn);
+        }
+    }
+
+    private doHandleTimeGraphResetSignal() {
+        this.unitController.viewRange = { start: 0, end: this.unitController.absoluteRange };
+    }
+
+    private zoomButton(zoomIn: boolean) {
+        let newStartRange = 0;
+        let newEndRange = 0;
+        let position = (this.unitController.viewRange.end + this.unitController.viewRange.start) / 2;
+        const percentZoom = 0.9;
+        if (this.unitController.selectionRange) {
+            if (this.unitController.selectionRange.end < this.unitController.selectionRange.start){
+                const temp = this.unitController.selectionRange.end;
+                this.unitController.selectionRange.end = this.unitController.selectionRange.start;
+                this.unitController.selectionRange.start = temp;
+            }
+            if (this.unitController.selectionRange.start !== this.unitController.selectionRange.end) {
+                position = (this.unitController.selectionRange.end + this.unitController.selectionRange.start) / 2;
+            }
+            if (this.unitController.selectionRange.start > this.unitController.viewRange.end || this.unitController.selectionRange.end < this.unitController.viewRange.start) {
```

Comment:
The position is set at line 225, then possibly changed at 234. If this condition is true it resets the position to the 225 value. Just use this condition to prevent the change at 234?

Reply:
How it's done in TC is to check if the middle time between selection start and selection end is within the view range. That way it doesn't matter if start < end or start > end.